src/main/java/com/github/curriculeon/People.java

```diff
@@ -0,0 +1,59 @@
+package com.github.curriculeon;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+public abstract class People <E extends Person> implements  Iterable<E>  {
+    private List<E> personList = new ArrayList<E>();
+
+    public void add(E person){
+        personList.add(person);
+    }
+
+    public E findById(long id){
+        for(E person: this.personList){
+            if(person.id == id)
+                return person;
+        }
+        return null;
+    }
+    public boolean contains(E person){
+        return personList.contains(person);
+    }
+    public void remove(E person) {
+        personList.remove(person);
+    }
+
+    public void remove(long id){
+        for(E person: personList){
+            if(person.id == id) {
+                personList.remove(person);
+                return;
+            }
+        }
+    }
+    public void removeAll(){
+        this.personList = null;
```

Comment:
This will ensure that you cannot `add` after removing all.
Try instead
`this.personList = new ArrayList<>();`
or
`this.personList.clear();`